frontend/packages/ux-editor/src/components/config/editModal/EditComponentId.tsx

```diff
@@ -16,45 +14,39 @@ export const EditComponentId = ({
   handleComponentUpdate,
   helpText,
 }: IEditComponentId) => {
-  const { components, containers } = useSelectedFormLayout();
-  const { t } = useTranslation();
-
   const handleIdChange = (id: string) => {
     handleComponentUpdate({
       ...component,
       id,
     });
   };
 
+  // TODO
+  // Missing:
+  // - helpText
+  // - Validation/Error handling
```

Comment:
Off the top of my head, I think I would add a prop to the `StudioTextField` component called `customValidationRules` or something similar, that takes a `string` as input, and returns an error message (or null/undefined if there is no error).

In `StudioTextField`, I would then call this validation function, and if there is an error message returned:
- keep the text field open for editing, and
- pass the error message to the `TextField` to display it.

You can look at the `customValidationRules` in the diff for this file here, and implement that as the custom validation function.
However, that only covers a part of the validation.

The `FormField` component handels validation against the component JSON schema. This validation will not be performed by default now. 
@framitdavid any thoughts on how to solve this? Should we extend the `StudioTextField` component to accept a jsonschema and perform validation against that?

Reply:
Good suggestion, @nkylstad! Should we create two separate components for this purpose? One with schema validation and one without? The one with schema validation can leverage the one without validation to expand support for validation against the schema. This way, we'll be able to segment the code for easier maintenance, and we'll have two separate components: one for schema validation and one without. What do you think? 🤔